Exercise_1.java

```diff
@@ -6,28 +6,50 @@
     boolean isEmpty() 
     { 
         //Write your code here 
+    	if(top == -1) {
+    		System.out.println("Stack empty");
+    		return true;
+    	}
+    	return false;
     } 
 
     Stack() 
     { 
         //Initialize your constructor 
+    	top = -1;
     } 
   
     boolean push(int x) 
     { 
         //Check for stack Overflow
+    	if(top > MAX-1) {

```

Comment:
The StackOverFlow would occur at "if(top == MAX-1)". Please go through the logic once again.

Reply:
Thanks for the feedback Samarth.
The overflow occurs if top >= MAX-1 ??

Reply:
Yes.